lib/sdf-server/src/service/v2/func/binding/attribute/reset_attribute_binding.rs

```diff
@@ -51,15 +52,42 @@ pub async fn reset_attribute_binding(
         )
         .await?;
 
-        if let EventualParent::SchemaVariant(schema_variant_id) = eventual_parent {
-            let schema =
-                SchemaVariant::schema_id_for_schema_variant_id(&ctx, schema_variant_id).await?;
-            let schema_variant = SchemaVariant::get_by_id_or_error(&ctx, schema_variant_id).await?;
-
-            WsEvent::schema_variant_updated(&ctx, schema, schema_variant)
-                .await?
-                .publish_on_commit(&ctx)
+        match eventual_parent {
+            EventualParent::SchemaVariant(schema_variant_id) => {
+                let schema =
+                    SchemaVariant::schema_id_for_schema_variant_id(&ctx, schema_variant_id).await?;
+                let schema_variant =
+                    SchemaVariant::get_by_id_or_error(&ctx, schema_variant_id).await?;
+                ctx.write_audit_log(
+                    AuditLogKind::DetachFunc {
+                        func_id,
+                        func_display_name: func.display_name.clone(),
+                        schema_variant_id: Some(schema_variant_id),
+                        component_id: None,
+                        subject_name: schema_variant.display_name().to_owned(),
+                    },
+                    func.name.clone(),
+                )
+                .await?;
+                WsEvent::schema_variant_updated(&ctx, schema, schema_variant)
+                    .await?
+                    .publish_on_commit(&ctx)
+                    .await?;
+            }
+            EventualParent::Component(component_id) => {
+                let component = Component::get_by_id(&ctx, component_id).await?;
+                ctx.write_audit_log(
+                    AuditLogKind::DetachFunc {
+                        func_id,
+                        func_display_name: func.display_name.clone(),
+                        schema_variant_id: None,
+                        component_id: Some(component_id),
+                        subject_name: component.name(&ctx).await?.to_owned(),
+                    },
+                    func.name.clone(),
+                )
                 .await?;
+            }
```

Comment:
Great use of breaking out an `if let` into a `match`... sometimes you'll see uses of `match` when unnecessary to expose all the variants left unused to avoid bugs if more variants are added to the enum. Example:
```rust
enum MyEnum {
    One
    Two
    Three
}

impl MyEnum {
    pub fn name(&self) -> &'static str {
        match self {
            One => "one"
            Two => "one"
            Three => "one"
        }
    }
}
```

By using a `match`, we cover all the variants by default unless we use a wildcard, variable or match guard.

If that function instead used `if let One = self { // code } else { // other code }`, then adding new variants could cause undesired behavior in the `else` block.

I bring this all up since this is a great example of a sneaky `if let` that might not have been obvious from static code analysis, but you found it for this PR. Nice!